src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -5153,27 +5164,16 @@ export class ResultsViewPageStore {
     > = remoteData({
         await: () => [this.cbioportalMutationCountData],
         invoke: () => {
-            const countsMap = _.groupBy(
-                this.cbioportalMutationCountData.result!,
-                count => mutationCountByPositionKey(count)
-            );
             return Promise.resolve((mutation: Mutation): number => {
                 const key = mutationCountByPositionKey(mutation);
-                const counts = countsMap[key];
-                if (counts) {
-                    return counts.reduce((count, next) => {
-                        return count + next.count;
-                    }, 0);
-                } else {
-                    return -1;
-                }
+                return this.cbioportalMutationCountData.result![key] || -1;
             });
         },
     });
     //COSMIC count
-    readonly cosmicCountData = remoteData<CosmicMutation[]>({
+    readonly cosmicCountData = remoteData<ComplexKeyCounter>({
```

Comment:
i think maybe this name should reflect it's somewhat involved typing better?   so we can more easily understand what the nature of the complex key is?

Reply:
you mean like `indexedCosmicCounts`? `cosmicCountsByKeywordAndPosition`?

Reply:
right